src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            return message => message.Match()
+                .With<ReplayedAllMessage>(replayed =>
+                {
+                    var seqNr = replayed.Persistent.SequenceNr;
+                    Buffer.Add(new EventEnvelope(
+                        offset: seqNr,
+                        persistenceId: replayed.Persistent.PersistenceId,
+                        sequenceNr: seqNr,
+                        @event: replayed.Persistent.Payload));
+                    CurrentOffset = replayed.Offset;
+                    Buffer.DeliverBuffer(TotalDemand);
+                })
+                .With<RecoverySuccess>(success =>
+                {
+                    Log.Debug("replay completed for current offset [{0}]", CurrentOffset);
+                    ReceiveRecoverySuccess(success.HighestSequenceNr);
+                })
+                .With<ReplayMessagesFailure>(failure =>
+                {
+                    Log.Debug("replay failed for offset [{0}], due to [{1}]", CurrentOffset, failure.Cause.Message);
+                    Buffer.DeliverBuffer(TotalDemand);
+                    OnErrorThenStop(failure.Cause);
+                })
+                .With<Request>(_ => Buffer.DeliverBuffer(TotalDemand))
+                .With<EventsByPersistenceIdPublisher.Continue>(() => { }) // skip during replay
+                .With<EventAppended>(() => { }) // skip during replay
+                .With<Cancel>(_ => Context.Stop(Self))
+                .WasHandled;
+        }
+    }
+
+    internal sealed class LiveAllEventsPublisher : AbstractAllEventsPublisher
+    {
+        private readonly ICancelable _tickCancelable;
+        public LiveAllEventsPublisher(long fromOffset, long toOffset, int maxBufferSize, string writeJournalPluginId, TimeSpan refreshInterval) 
+            : base(fromOffset, toOffset, maxBufferSize, writeJournalPluginId)
+        {
+            _tickCancelable = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(refreshInterval, refreshInterval, Self, AllEventsPublisher.Continue.Instance, Self);
+        }
+
+        protected override void PostStop()
+        {
+            _tickCancelable.Cancel();
+            base.PostStop();
+        }
+
+        protected override void ReceiveInitialRequest()
+        {
+            JournalRef.Tell(SubscribeAllEvents.Instance);
+            Replay();
+        }
+
+        protected override void ReceiveIdleRequest()
+        {
+            Buffer.DeliverBuffer(TotalDemand);
+            if (Buffer.IsEmpty && CurrentOffset > ToOffset)
+                OnCompleteThenStop();
+        }
+
+        protected override void ReceiveRecoverySuccess(long highestSequenceNr)
+        {
+            Buffer.DeliverBuffer(TotalDemand);
+            if (Buffer.IsEmpty && CurrentOffset > ToOffset)
+                OnCompleteThenStop();
+
+            Context.Become(Idle);
+        }
+    }
+
+    internal sealed class CurrentAllEventsPublisher : AbstractAllEventsPublisher
```

Comment:
I don't like this naming. Maybe we could use simpler `EventsQuery` and `CurrentEventsQuery`?